Intermittent test failure in Spanish queue

See https://dev.azure.com/dnceng/public/_build/results?buildId=475513&view=ms.vss-test-web.build-test-results-tab&runId=15125436&resultId=104047&paneView=debug

**Test Name:** `ArgumentNullException_NoArguments_Warns`

**Error Message:**
```
Microsoft.CodeAnalysis.Testing.Verifiers.EqualWithMessageException : Context: Diagnostics of test state\r\nExpected diagnostic message to be "Call the ArgumentNullException constructor that contains a message and/or paramName parameter." was "Llame al constructor ArgumentNullException que contiene un mensaje o un parámetro paramName."\r\n\r\nDiagnostic:\r\n // Test0.vb(4,31): warning CA2208: Llame al constructor ArgumentNullException que contiene un mensaje o un parámetro paramName.\r\nGetBasicResultAt(4, 31, InstantiateArgumentExceptionsCorrectlyAnalyzer.CA2208)\r\n\r\n\r\nAssert.Equal() Failure\r\nExpected: Call the ArgumentNullException constructor that contains a message and/or paramName parameter.\r\nActual: Llame al constructor ArgumentNullException que contiene un mensaje o un parámetro paramName.
```

**Stack:**
```
at Microsoft.CodeAnalysis.Testing.Verifiers.XUnitVerifier.Equal[T](T expected, T actual, String message) in /_/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Testing.Verifiers.XUnit/XUnitVerifier.cs:line 48
   at Microsoft.CodeAnalysis.Testing.AnalyzerTest`1.VerifyDiagnosticResults(IEnumerable`1 actualResults, ImmutableArray`1 analyzers, DiagnosticResult[] expectedResults, IVerifier verifier) in /_/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs:line 316
   at Microsoft.CodeAnalysis.Testing.AnalyzerTest`1.<VerifyDiagnosticsAsync>d__55.MoveNext() in /_/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs:line 197
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.Testing.CodeFixTest`1.<RunAsync>d__38.MoveNext() in /_/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.CodeFix.Testing/CodeFixTest`1.cs:line 189
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Test.Utilities.VisualBasicCodeFixVerifier`2.<VerifyAnalyzerAsync>d__4.MoveNext() in F:\workspace\_work\1\s\src\Test.Utilities\VisualBasicCodeFixVerifier`2.cs:line 34
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.NetCore.Analyzers.Runtime.UnitTests.InstantiateArgumentExceptionsCorrectlyTests.<ArgumentNullException_NoArguments_Warns>d__9.MoveNext() in F:\workspace\_work\1\s\src\NetAnalyzers\UnitTests\Microsoft.NetCore.Analyzers\Runtime\InstantiateArgumentExceptionsCorrectlyTests.cs:line 172
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
```

--- a/src/Microsoft.NetCore.Analyzers/Core/Runtime/InstantiateArgumentExceptionsCorrectly.cs
+++ b/src/Microsoft.NetCore.Analyzers/Core/Runtime/InstantiateArgumentExceptionsCorrectly.cs
@@ -1,7 +1,6 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.Collections.Immutable;
-using System.Globalization;
 using Analyzer.Utilities;
 using Analyzer.Utilities.Extensions;
 using Microsoft.CodeAnalysis;
@@ -26,16 +25,36 @@ namespace Microsoft.NetCore.Analyzers.Runtime
         private static readonly LocalizableString s_localizableMessageIncorrectParameterName = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyMessageIncorrectParameterName), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
         private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
 
-        internal static DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(RuleId,
+        internal static DiagnosticDescriptor RuleNoArguments = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
-                                                                             "{0}",
+                                                                             s_localizableMessageNoArguments,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticHelpers.DefaultDiagnosticSeverity,
                                                                              isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
                                                                              description: s_localizableDescription,
                                                                              helpLinkUri: HelpUri,
                                                                              customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+
+        internal static DiagnosticDescriptor RuleIncorrectMessage = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageIncorrectMessage,
+                                                                             DiagnosticCategory.Usage,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: HelpUri,
+                                                                             customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        internal static DiagnosticDescriptor RuleIncorrectParameterName = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessageIncorrectParameterName,
+                                                                             DiagnosticCategory.Usage,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: HelpUri,
+                                                                             customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleNoArguments, RuleIncorrectMessage, RuleIncorrectParameterName);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
@@ -78,7 +97,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime
                 if (HasMessageOrParameterNameConstructor(creation.Type))
                 {
                     // Call the {0} constructor that contains a message and/ or paramName parameter
-                    ReportDiagnostic(context, s_localizableMessageNoArguments, creation.Type.Name);
+                    context.ReportDiagnostic(context.Operation.Syntax.CreateDiagnostic(RuleNoArguments, creation.Type.Name));
                 }
             }
             else
@@ -108,11 +127,11 @@ namespace Microsoft.NetCore.Analyzers.Runtime
             OperationAnalysisContext context)
         {
             bool matchesParameter = MatchesParameter(targetSymbol, creation, stringArgument);
-            LocalizableString? format = null;
+            DiagnosticDescriptor? rule = null;
 
             if (IsMessage(parameter) && matchesParameter)
             {
-                format = s_localizableMessageIncorrectMessage;
+                rule = RuleIncorrectMessage;
             }
             else if (IsParameterName(parameter) && !matchesParameter)
             {
@@ -122,21 +141,13 @@ namespace Microsoft.NetCore.Analyzers.Runtime
                     return;
                 }
 
-                format = s_localizableMessageIncorrectParameterName;
+                rule = RuleIncorrectParameterName;
             }
 
-            if (format != null)
+            if (rule != null)
             {
-                ReportDiagnostic(context, format, targetSymbol.Name, stringArgument, parameter.Name, creation.Type.Name);
+                context.ReportDiagnostic(context.Operation.Syntax.CreateDiagnostic(rule, targetSymbol.Name, stringArgument, parameter.Name, creation.Type.Name));
             }
-        }
-
-        private static void ReportDiagnostic(OperationAnalysisContext context, LocalizableString format, params object[] args)
-        {
-            context.ReportDiagnostic(
-                context.Operation.Syntax.CreateDiagnostic(
-                    Descriptor,
-                    string.Format(CultureInfo.CurrentCulture, format.ToString(CultureInfo.CurrentCulture), args)));
         }
 
         private static bool IsMessage(IParameterSymbol parameter)

--- a/src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/InstantiateArgumentExceptionsCorrectlyTests.cs
+++ b/src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/InstantiateArgumentExceptionsCorrectlyTests.cs
@@ -1,6 +1,5 @@
 // Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System.Globalization;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Testing;
 using Test.Utilities;
@@ -16,10 +15,6 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
 {
     public class InstantiateArgumentExceptionsCorrectlyTests
     {
-        private static readonly string s_noArguments = MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyMessageNoArguments;
-        private static readonly string s_incorrectMessage = MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyMessageIncorrectMessage;
-        private static readonly string s_incorrectParameterName = MicrosoftNetCoreAnalyzersResources.InstantiateArgumentExceptionsCorrectlyMessageIncorrectParameterName;
-
         [Fact]
         public async Task ArgumentException_NoArguments_Warns()
         {
@@ -31,7 +26,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentException();
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_noArguments, "ArgumentException"));
+                GetCSharpNoArgumentsExpectedResult(6, 31, "ArgumentException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -39,7 +34,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentException()
                     End Sub
                 End Class",
-              GetBasicExpectedResult(4, 31, s_noArguments, "ArgumentException"));
+              GetBasicNoArgumentsExpectedResult(4, 31, "ArgumentException"));
         }
 
         [Fact]
@@ -53,7 +48,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentNullException("""");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "", "paramName", "ArgumentNullException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "", "paramName", "ArgumentNullException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -61,7 +56,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentNullException("""")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", "", "paramName", "ArgumentNullException"));
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", "", "paramName", "ArgumentNullException"));
         }
 
         [Fact]
@@ -75,7 +70,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentNullException("" "");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", " ", "paramName", "ArgumentNullException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", " ", "paramName", "ArgumentNullException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -83,7 +78,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentNullException("" "")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", " ", "paramName", "ArgumentNullException"));
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", " ", "paramName", "ArgumentNullException"));
         }
 
         [Fact]
@@ -98,7 +93,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentNullException(nameof(foo));
                     }
                 }",
-                GetCSharpExpectedResult(7, 31, s_incorrectParameterName, "Test", "foo", "paramName", "ArgumentNullException"));
+                GetCSharpIncorrectParameterNameExpectedResult(7, 31, "Test", "foo", "paramName", "ArgumentNullException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -107,7 +102,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentNullException(NameOf(foo))
                     End Sub
                 End Class",
-                GetBasicExpectedResult(5, 31, s_incorrectParameterName, "Test", "foo", "paramName", "ArgumentNullException"));
+                GetBasicIncorrectParameterNameExpectedResult(5, 31, "Test", "foo", "paramName", "ArgumentNullException"));
         }
 
         [Fact]
@@ -121,7 +116,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentException(""first"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentException"));
+                GetCSharpIncorrectMessageExpectedResult(6, 31, "Test", "first", "message", "ArgumentException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -129,7 +124,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentException(""first"")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentException"));
+                GetBasicIncorrectMessageExpectedResult(4, 31, "Test", "first", "message", "ArgumentException"));
         }
 
         [Fact]
@@ -143,8 +138,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentException(""first"", ""first is incorrect"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentException"),
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "first is incorrect", "paramName", "ArgumentException"));
+                GetCSharpIncorrectMessageExpectedResult(6, 31, "Test", "first", "message", "ArgumentException"),
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "first is incorrect", "paramName", "ArgumentException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -152,8 +147,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentException(""first"", ""first is incorrect"")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentException"),
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", "first is incorrect", "paramName", "ArgumentException"));
+                GetBasicIncorrectMessageExpectedResult(4, 31, "Test", "first", "message", "ArgumentException"),
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", "first is incorrect", "paramName", "ArgumentException"));
         }
 
         [Fact]
@@ -167,7 +162,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentNullException();
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_noArguments, "ArgumentNullException"));
+                GetCSharpNoArgumentsExpectedResult(6, 31, "ArgumentNullException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -175,7 +170,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentNullException()
                     End Sub
                 End Class",
-                 GetBasicExpectedResult(4, 31, s_noArguments, "ArgumentNullException"));
+                 GetBasicNoArgumentsExpectedResult(4, 31, "ArgumentNullException"));
         }
 
         [Fact]
@@ -189,7 +184,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentNullException(""first is null"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "first is null", "paramName", "ArgumentNullException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "first is null", "paramName", "ArgumentNullException"));
         }
 
         [Fact]
@@ -203,8 +198,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentNullException(""first is null"", ""first"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "first is null", "paramName", "ArgumentNullException"),
-                GetCSharpExpectedResult(6, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentNullException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "first is null", "paramName", "ArgumentNullException"),
+                GetCSharpIncorrectMessageExpectedResult(6, 31, "Test", "first", "message", "ArgumentNullException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -212,8 +207,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentNullException(""first is null"", ""first"")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", "first is null", "paramName", "ArgumentNullException"),
-                GetBasicExpectedResult(4, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentNullException"));
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", "first is null", "paramName", "ArgumentNullException"),
+                GetBasicIncorrectMessageExpectedResult(4, 31, "Test", "first", "message", "ArgumentNullException"));
         }
 
         [Fact]
@@ -227,7 +222,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentOutOfRangeException();
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_noArguments, "ArgumentOutOfRangeException"));
+                GetCSharpNoArgumentsExpectedResult(6, 31, "ArgumentOutOfRangeException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -235,7 +230,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentOutOfRangeException()
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_noArguments, "ArgumentOutOfRangeException"));
+                GetBasicNoArgumentsExpectedResult(4, 31, "ArgumentOutOfRangeException"));
         }
 
         [Fact]
@@ -249,7 +244,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentOutOfRangeException(""first is out of range"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -257,7 +252,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentOutOfRangeException(""first is out of range"")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"));
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"));
         }
 
         [Fact]
@@ -271,8 +266,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.ArgumentOutOfRangeException(""first is out of range"", ""first"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"),
-                GetCSharpExpectedResult(6, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentOutOfRangeException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"),
+                GetCSharpIncorrectMessageExpectedResult(6, 31, "Test", "first", "message", "ArgumentOutOfRangeException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -280,8 +275,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.ArgumentOutOfRangeException(""first is out of range"", ""first"")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"),
-                GetBasicExpectedResult(4, 31, s_incorrectMessage, "Test", "first", "message", "ArgumentOutOfRangeException"));
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", "first is out of range", "paramName", "ArgumentOutOfRangeException"),
+                GetBasicIncorrectMessageExpectedResult(4, 31, "Test", "first", "message", "ArgumentOutOfRangeException"));
         }
 
         [Fact]
@@ -295,7 +290,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.DuplicateWaitObjectException();
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_noArguments, "DuplicateWaitObjectException"));
+                GetCSharpNoArgumentsExpectedResult(6, 31, "DuplicateWaitObjectException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -303,7 +298,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.DuplicateWaitObjectException()
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_noArguments, "DuplicateWaitObjectException"));
+                GetBasicNoArgumentsExpectedResult(4, 31, "DuplicateWaitObjectException"));
         }
 
         [Fact]
@@ -317,7 +312,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.DuplicateWaitObjectException(""first is duplicate"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -325,7 +320,7 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.DuplicateWaitObjectException(""first is duplicate"")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"));
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"));
         }
 
         [Fact]
@@ -339,8 +334,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         throw new System.DuplicateWaitObjectException(""first is duplicate"", ""first"");
                     }
                 }",
-                GetCSharpExpectedResult(6, 31, s_incorrectParameterName, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"),
-                GetCSharpExpectedResult(6, 31, s_incorrectMessage, "Test", "first", "message", "DuplicateWaitObjectException"));
+                GetCSharpIncorrectParameterNameExpectedResult(6, 31, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"),
+                GetCSharpIncorrectMessageExpectedResult(6, 31, "Test", "first", "message", "DuplicateWaitObjectException"));
 
             await VerifyVB.VerifyAnalyzerAsync(@"
                 Public Class [MyClass]
@@ -348,8 +343,8 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                         Throw New System.DuplicateWaitObjectException(""first is duplicate"", ""first"")
                     End Sub
                 End Class",
-                GetBasicExpectedResult(4, 31, s_incorrectParameterName, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"),
-                GetBasicExpectedResult(4, 31, s_incorrectMessage, "Test", "first", "message", "DuplicateWaitObjectException"));
+                GetBasicIncorrectParameterNameExpectedResult(4, 31, "Test", "first is duplicate", "parameterName", "DuplicateWaitObjectException"),
+                GetBasicIncorrectMessageExpectedResult(4, 31, "Test", "first", "message", "DuplicateWaitObjectException"));
         }
 
 
@@ -611,20 +606,34 @@ namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
                 }");
         }
 
-        private static DiagnosticResult GetCSharpExpectedResult(int line, int column, string format, params string[] args)
-        {
-            string message = string.Format(CultureInfo.CurrentCulture, format, args);
-            return new DiagnosticResult(InstantiateArgumentExceptionsCorrectlyAnalyzer.Descriptor)
+        private static DiagnosticResult GetCSharpNoArgumentsExpectedResult(int line, int column, string typeName) =>
+            VerifyCS.Diagnostic(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleNoArguments)
                 .WithLocation(line, column)
-                .WithMessage(message);
-        }
+                .WithArguments(typeName);
 
-        private static DiagnosticResult GetBasicExpectedResult(int line, int column, string format, params string[] args)
-        {
-            string message = string.Format(CultureInfo.CurrentCulture, format, args);
-            return new DiagnosticResult(InstantiateArgumentExceptionsCorrectlyAnalyzer.Descriptor)
+        private static DiagnosticResult GetBasicNoArgumentsExpectedResult(int line, int column, string typeName) =>
+            VerifyVB.Diagnostic(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleNoArguments)
                 .WithLocation(line, column)
-                .WithMessage(message);
-        }
+                .WithArguments(typeName);
+
+        private static DiagnosticResult GetCSharpIncorrectMessageExpectedResult(int line, int column, params string[] args) =>
+            VerifyCS.Diagnostic(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleIncorrectMessage)
+                .WithLocation(line, column)
+                .WithArguments(args);
+
+        private static DiagnosticResult GetBasicIncorrectMessageExpectedResult(int line, int column, params string[] args) =>
+            VerifyVB.Diagnostic(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleIncorrectMessage)
+                .WithLocation(line, column)
+                .WithArguments(args);
+
+        private static DiagnosticResult GetCSharpIncorrectParameterNameExpectedResult(int line, int column, params string[] args) =>
+            VerifyCS.Diagnostic(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleIncorrectParameterName)
+                .WithLocation(line, column)
+                .WithArguments(args);
+
+        private static DiagnosticResult GetBasicIncorrectParameterNameExpectedResult(int line, int column, params string[] args) =>
+            VerifyVB.Diagnostic(InstantiateArgumentExceptionsCorrectlyAnalyzer.RuleIncorrectParameterName)
+                .WithLocation(line, column)
+                .WithArguments(args);
     }
 }

